### PR TITLE
Run PingSource and Kafka Broker tests 

### PR DESCRIPTION
**Pull request by pierDipi**
Run tests for:

- PingSource
- KafkaSource
- Kafka Broker

Ignore changes in vendor and third_party since those are autogenerated by running `hack/update-deps --upgrade`

### CODE REVIEW

File: hack/install-dependencies.sh [hunk 1]
```diff
@@ -52,7 +51,12 @@ kubectl patch deployment \
...
   --patch-file "${REPO_ROOT_DIR}/hack/eventing-injection-disabled.yaml"
 
-"${REPO_ROOT_DIR}"/third_party/kafka/kafka_setup.sh
+"${REPO_ROOT_DIR}"/third_party/eventing-kafka-broker/test/kafka/kafka_setup.sh
+
+kubectl apply -n "${ISTIO_NAMESPACE}" -Rf "${REPO_ROOT_DIR}"/test/config
+kubectl apply -n "${SYSTEM_NAMESPACE}" -Rf "${REPO_ROOT_DIR}"/test/config
 
-# curl -X POST -v -H "content-type: application/json" -H "ce-specversion: 1.0" -H "ce-source: my/curl/command" -H "ce-type: my.demo.event" -H "ce-id: 0815" -d '{"value":"Hello Knative"}' http://broker-ingress.knative-eventing.svc.cluster.local/test/default
+source "${REPO_ROOT_DIR}/third_party/eventing-kafka-broker/test/e2e-common.sh"
```
Comment (matzew) [inline hack/install-dependencies.sh:59]: nice streamlining this

File: test/config/destination-policy.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Disable mTLS for connections to the API server
+apiVersion: networking.istio.io/v1beta1
+kind: DestinationRule
+metadata:
```
Comment (matzew) [inline test/config/destination-policy.yaml:4]: where do we apply?
do we need to system namespace?

File: test/e2e-common.sh [hunk 3]
```diff
@@ -1,14 +1,48 @@
...
 function knative_setup() {
-  "${REPO_ROOT_DIR}"/hack/update-istio.sh || return $?
+  git submodule update --init --recursive
   "${REPO_ROOT_DIR}"/hack/install-dependencies.sh || return $?
 }
+
+function run_eventing_core_tests() {
+  pushd "${REPO_ROOT_DIR}"/third_party/eventing || return $?
+
+  export BROKER_TEMPLATES="${REPO_ROOT_DIR}/test/e2e/templates/kafka-broker"
```
Comment (matzew) [inline test/e2e-common.sh:21]: where is the template dir used?

File: test/config/destination-policy.yaml (hunk 2 above)
Comment (pierDipi) [reply test/config/destination-policy.yaml:4]: it's applied in both istio-system and knative-eventing as part of install-dependencies.sh

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**